Let us consider a hydrogel fuel cell. The reaction is:

$$H_2(g) + (1/2)O_2(g) \rightarrow H_2O(g) $$

The rate at which water vapor is generated is $\bar{\dot{n}}_\text{p,H2O}$. The number of electrons produced per molecule of the product is $n_e$. The rate that electrons are delivered by the cell to the load is:

$$n_e N_0 \bar{\dot{n}}_\text{p,H2O}$$

The current is then:

$$I = q n_e N_0 \bar{\dot{n}}_\text{p,H2O} $$

For hydrogen/oxygen cells, $n_e = 2$.

Recall that the theoretical efficiency of a reversible fuel cell is:

$$\eta_\text{rev} = \frac{\Delta G}{\Delta H} $$

The efficiency of practical fuel cells is the ratio of the electrical power, $P_\text{L}$, delivered to the load to the heat power that would be generated by combining the reactants under the same temperature and pressure used in the cell:

$$\eta_\text{practical} = \frac{P_\text{L}}{P_\text{in}} = \frac{I_\text{L} V_\text{L}}{\Delta \bar{h}_\text{comb} \bar{\dot{n}}_\text{p,H2O}}$$

Substituting the equation for current, assuming 100% current efficiency ($I_L = I$), yields:

$$\eta_\text{practical} = \frac{P_\text{L}}{P_\text{in}} = \frac{(q n_e N_0 \bar{\dot{n}}_\text{p,H2O}) V_\text{L}}{\Delta \bar{h}_\text{comb} \bar{\dot{n}}_\text{p,H2O}} = \frac{q n_e N_0  V_\text{L}}{\Delta \bar{h}_\text{comb} }$$


In [6]:
hfo_H2_kJperKmol = 0
hfo_O2_kJperKmol = 0
hfo_H2O_gas_kJperKmol = -241820

# enthalpy of the reaction per kmol of H2
h_C_HHV_kJperKmolH2 = hfo_H2O_gas_kJperKmol - (hfo_O2_kJperKmol/2) - hfo_H2_kJperKmol; print(h_C_HHV_kJperKmolH2)

q_out_kJperKmolH2 = - h_C_HHV_kJperKmolH2; print(q_out_kJ)

-241820.0
241820.0


In [7]:
ne = 2 
N0_moleculesPermol = 6.022e23
q_C = 1.6e-19
# h_C_HHV_JpermolH2 = h_C_HHV_kJperKmolH2
q_out_JpermolH2 = q_out_kJperKmolH2

# hVoltage_V = (h_C_HHV_JpermolH2)/(n_p_H2O_vap_mol*ne*N0_moleculesPermol*q_C); print(hVoltage_V)

hVoltage_V = q_out_JpermolH2/(q_C * ne * N0_moleculesPermol); print(hVoltage_V)

1.254877947525739


In [9]:
# The inverse of the enthalpy voltage is then
print(1/hVoltage_V)

0.7968902489454966


The practical efficiency is then:

$$\eta_\text{practical} = 0.797 V_L $$